SeQuant/core/optimize/fusion.cpp

```diff
@@ -102,4 +104,29 @@ ExprPtr Fusion::fuse_right(Product const& lhs, Product const& rhs) {
   return ex<Product>(ExprPtrList{ex<Sum>(ExprPtrList{a, b}), f});
 }
 
+rational Fusion::gcd_rational(rational const& left, rational const& right) {
+  auto&& r1 = left.real();
+  auto&& r2 = right.real();
+  auto&& n1 = numerator(r1);
+  auto&& d1 = denominator(r1);
+  auto&& n2 = numerator(r2);
+  auto&& d2 = denominator(r2);
```

Comment:
These should be `const auto &` - forwarding reference are only useful in combination with `std::forward`, which in turn is only useful if data needs to be copied around.